sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -21,149 +21,54 @@
 # SOFTWARE.
 
 
-
 """Http Logging Policy for Azure SDK"""
 
-import os
-import urllib
+import json
 import logging
-import types
-import ast
+import time
+from typing import Optional
 
+from azure.core.pipeline import PipelineRequest, PipelineResponse, HTTPRequestType, HTTPResponseType
 from azure.core.pipeline.policies import HttpLoggingPolicy
+from .http_constants import HttpHeaders
 
-
-
+def _format_error(payload: str) -> str:
+    output = json.loads(payload)
+    return output['message'].replace("\r", " ")
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs): # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
-        if self._enable_diagnostics_logging:
-            self.logger = logger
+        self.logger = logger or logging.getLogger(
+            "azure.cosmos._cosmos_http_logging_policy"
```

Comment:
I'm not sure we should overwrite `self.logger` - this is set in Core as "azure.core.pipeline.policies.http_logging_policy", and if we reset that, it will be slightly more difficult for customers to configure their logging filter.
If you _do_ want to do this, I would remove the private namespace and just use something like "azure.cosmos.http_logging_policy"

Reply:
Also - this (and all the following lines in the constructor) should only be done if `enable_diagnostis` is `True`